lib/recipes-data/src/lib/extract_recipedata_from_element.ts

```diff
@@ -0,0 +1,30 @@
+import type {Block} from "@guardian/content-api-models/v1/block";
+import {ElementType} from "@guardian/content-api-models/v1/elementType";
+
+
+export interface RecipeOutput {
+  recipeUId: string;
+  jsonBlob: string;
+  checksum?: string;
+}
+
+export function extractRecipeData(webUrl: string, block: Block): RecipeOutput[] {
+  const allRecipes = block.elements.filter(elem => elem.type === ElementType.RECIPE)
+  if (allRecipes.length != 0) {
+    return allRecipes.map(recp => parseJsonBlob(webUrl, recp.recipeTypeData?.recipeJson))
+  } else {
+    throw new Error("Error! No recipe is present in elements!")
```

Comment:
I don't think that this should be an error condition; I think simply returning an empty array is OK.
(Remember, most of the articles we will be operating on have no recipes in them)

In that case you don't even need the `if`/`else`; you can just `return allRecipes.map(...)` and that will give you an empty output array if the input was empty.